File: VaultNetAPI/VaultNet/Db/Product.cs
using ChargedChat.Db;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VaultNet.Db
{
    public class Product
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int ProductId { get; set; }

        public int BussinessId { get; set; }

        [ForeignKey(nameof(BussinessId))]
        public Bussiness Bussiness { get; set; }

        [Required, StringLength(100)]
        public string Name { get; set; }

        public string Description { get; set; }

        public string Image {  get; set; }

        [Required]
        public int Quantity { get; set; }

        public int UnitId {  get; set; }

        [ForeignKey(nameof(UnitId))]
        public Unit Unit { get; set; }

        public int CategoryId { get; set; }

        [ForeignKey(nameof(CategoryId))]
        public Category Category { get; set; }


        public virtual ICollection<Change> Changes { get; set; }
        public virtual ICollection<Snapshot> Snapshots { get; set; }
    }
}

File: VaultNetAPI/src/main/java/com/ventthos/Vaultnet/domain/Product.java
package com.ventthos.Vaultnet.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Set;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long productId;

    @ManyToOne
    @JoinColumn(name = "business_id", nullable = false)
    private Business business;

    @Column(length = 100, nullable = false)
    private String name;

    private String description;

    private String image;

    private Integer quantity;

    @ManyToOne
    @JoinColumn(name = "unit_id", nullable = false)
    private Unit unit;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    private Category category;

    @ManyToMany
    @JoinTable(
            name = "product_snapshot",
            joinColumns = @JoinColumn(name = "product_id"),
            inverseJoinColumns = @JoinColumn(name = "snapshot_id")
    )
    private Set<Snapshot> snapshots;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL)
    private List<Change> changes;
}
